rpc/wrtc_server_stream.go

```diff
@@ -221,31 +221,34 @@ func (s *webrtcServerStream) SendMsg(m interface{}) (err error) {
 }
 
 func (s *webrtcServerStream) onRequest(request *webrtcpb.Request) {
+	// Error cases here are logged at the warn level. It's not a server error to find client
```

Comment:
This was found with blinky and already having 3-4 PRs open I decided to just lump these in here rather than make a new ticket.

Specifically, blinky got an error on [this assertion](https://github.com/viamrobotics/rdk/blob/a3262bfbf30c391a7f45804144e9d9184fc30d43/robot/impl/resource_manager_test.go#L1756).

The log that caused the failure seemed typical:
```
    wrtc_server_stream.go:243: 2024-11-11T21:02:13.100Z	ERROR	remote1.networking.grpc_requests	rpc/wrtc_server_stream.go:243	error closing	{"error":"io: read/write on closed pipe","method":"/grpc.reflection.v1alpha.ServerReflection/ServerReflectionInfo"}
```

Alternatively, we can remove that assertion. It's unclear to me what it was protecting against. But as I stated above, I didn't feel the desire to create a new ticket/branch/PR just to remove that assertion.

Reply:
Half strike this -- my first fix was incomplete in that the `closedReason` was now being accessed without a mutex.

Rather than moving that to another "atomic" value, I decided to just remove it. It was only being used to fill in log messages.